src/AutoRest.CSharp/Common/Output/Models/Types/SchemaObjectType.cs

```diff
@@ -150,7 +150,7 @@ protected override ObjectTypeConstructor BuildSerializationConstructor()
 
             return new ObjectTypeConstructor(
                 Type.Name,
-                IsAbstract ? Protected : Internal,
+                (IsAbstract || ObjectSchema?.Extensions?.MgmtTypeReferenceType == true) ? Protected : Internal,
```

Comment:
could we introduce a local variable for this flag and add some comment to explain why?

Reply:
do you have a draft pr in sdk repo?